hydra-task/src/main/java/com/addthis/hydra/task/map/MapFeeder.java

```diff
@@ -103,20 +103,23 @@ public MapFeeder(StreamMapper task, TaskDataSource source, int feeders) {
         }
 
         try {
+            boolean interrupt = false;
             if (source.isEnabled()) {
                 while (fillBuffer()) {
                     if (Thread.interrupted()) {
+                        interrupt = true;
                         closeSourceIfNeeded();
                     }
                 }
             }
+            final boolean interrupted = interrupt;
```

Comment:
it looks like there could be a race condition where the map feeder gets interrupted during many of the later steps after it runs out of bundles in the normal manner. eg. source is exhausted, loop exits, map feeder starts to close sources etc, (this takes a little while), interrupt is sent, map feeder finishes exiting without seeing interrupt.